src/validation.cpp

```diff
@@ -3408,6 +3408,7 @@ bool ProcessNewBlock(CValidationState& state, CNode* pfrom, const std::shared_pt
     LogPrintf("%s : ACCEPTED Block %ld in %ld milliseconds with size=%d\n", __func__, newHeight, GetTimeMillis() - nStartTime,
               GetSerializeSize(*pblock, SER_DISK, CLIENT_VERSION));
 
+    SyncWithValidationInterfaceQueue();
```

Comment:
This change isn't good, it's essentially reverting all the benefits introduced by the asynchronous processing of the validationinterface handlers in a background worker thread. Putting to sleep the main network messages processing thread until every validationinterface subscriber finishes processing the events at every block, blocking any new message receive/processing until every event callbacks finishes (and this means the handlers in the wallet, GUI, net_processing, zmq, etc).

So you know further how this works:
Every event that is being signaled to the validationinterface (`UpdatedBlockTip`, `BlockConnected`, `BlockDisconnected`,`TransactionAddedToMempool`, etc), is added to a tasks queue, and a background worker thread takes them, one by one, in a memory consistent way (ordered), calling each of the subscribed callbacks asynchronously from the signal originator. Letting the signal originator to continue running concurrently.

not sure what was the problem that made you add this but surely we can find another way of solving it.

Reply:
Yeah... I know :)
This test started failing as I introduced the notification interface, and removed the direct calls to `UpdatedBlockTip`, after your question.
This call here is obviously wrong. I just pushed it here few minutes  ago, to see if the problem was there, and indeed is (have no way of testing it locally, as here is passing fine). It wasn't meant to be left there of course.
But well, calling the sync only in the test, before getting the new list at the tip should be enough. Let's see if GA likes it.